src/Core/Services/UsernameGenerationService.cs

```diff
@@ -0,0 +1,250 @@
+            {
+                return DefaultGenerated;
+            }
+
+            var atIndex = adressedEmail.IndexOf("@");
+            if (atIndex < 1 || atIndex >= adressedEmail.Length - 1)
+            {
+                return adressedEmail;
+            }
+
+            var emailBeginning = adressedEmail.Substring(0, atIndex);
+            var emailEnding = string.Empty;
+
+            emailEnding = adressedEmail.Substring(atIndex + 1, adressedEmail.Length - (atIndex + 1));
+
+            if (options.PlusAddressedEmailType == UsernameEmailType.Random)
+            {
+                generatedString = await RandomStringAsync(8);
+            }
+            else
+            {
+                generatedString = options.EmailWebsite;
+            }
+
+            return emailBeginning + "+" + generatedString + "@" + emailEnding;
+        }
+
+        private async Task<string> GenerateCatchAllAsync(UsernameGenerationOptions options)
+        {
+            options.Merge(_defaultOptions);
+            var catchAllEmailDomain = options.CatchAllEmailDomain;
+            string generatedString = string.Empty;
+
+            if (string.IsNullOrWhiteSpace(catchAllEmailDomain))
+            {
+                return DefaultGenerated;
+            }
+
+            if (options.CatchAllEmailType == UsernameEmailType.Random)
+            {
+                generatedString = await RandomStringAsync(8);
+            }
+            else
+            {
+                generatedString = options.EmailWebsite;
+            }
+
+            return generatedString + "@" + catchAllEmailDomain;
+        }
+
+        private async Task<string> GenerateForwardedEmailAliasAsync(UsernameGenerationOptions options)
+        {
+            options.Merge(_defaultOptions);
+
+            switch (options.ServiceType)
+            {
+                case ForwardedEmailServiceType.AnonAddy:
+                    if (string.IsNullOrWhiteSpace(options.AnonAddyApiAccessToken) || string.IsNullOrWhiteSpace(options.AnonAddyDomainName))
+                    {
+                        return DefaultGenerated;
+                    }
+                    return await GetAnonAddyUsername(options.AnonAddyApiAccessToken, options.AnonAddyDomainName);
+
+                case ForwardedEmailServiceType.FirefoxRelay:
+                    if (string.IsNullOrWhiteSpace(options.FirefoxRelayApiAccessToken))
+                    {
+                        return DefaultGenerated;
+                    }
+                    return await GetFirefoxRelayUsername(options.FirefoxRelayApiAccessToken);
+
+                case ForwardedEmailServiceType.SimpleLogin:
+                    if (string.IsNullOrWhiteSpace(options.SimpleLoginApiKey))
+                    {
+                        return DefaultGenerated;
+                    }
+                    return await GetSimpleLoginUsername(options.SimpleLoginApiKey);
+                default:
+                    return DefaultGenerated;
```

Comment:
call the logger to track that there is one service type that has not been implemented and then return the default